### PR TITLE
[WebUI] Add EventLogger class

### PR DESCRIPTION
**Pull request by caisq**
- The EventLogger logs user events in the app such as starting the app, switching between different tabs of the app, keypresses (without text content), abbreviation expansion triggers, word replacements, spelling for abbreviation expansion, TTS output and keystroke injection, adding, removing and selecting quick phrases, and changing settings, etc.
- Including the EventLogger abstract interface and HTTP based
  implementation
- Helper methods are included in events-logger-impl.ts
- Unit tests are added

### CODE REVIEW

**Comment by caisq** [comment]
@antonkast-google Please review this PR. Thanks.

File: webui/src/app/event-logger/event-logger-impl.ts [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+      return of({});
+    }
+    this.ensureUserIdSet();
+    const httpOptions = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json',
+      }),
+    };
+    return this.http.post<EventLogResponse>(
+        EVENT_LOGS_ENDPOINT, JSON.stringify(eventLogEntry), httpOptions);
```
Comment (antonkast-google) [inline webui/src/app/event-logger/event-logger-impl.ts:501]: Comments in email.

File: webui/src/app/quick-phrases/quick-phrases.component.ts [hunk 2]
```diff
@@ -197,6 +198,7 @@ export class QuickPhrasesComponent implements AfterViewInit, OnChanges,
       injectedKeys.push(VIRTUAL_KEY.SPACE);  // Append a space at the end.
       injectKeys(injectedKeys);
     }
+    // TODO(cais): Call this.eventLogger.logContextualPhraseSelection()
```
Comment (antonkast-google) [inline webui/src/app/quick-phrases/quick-phrases.component.ts:201]: Did you mean to do this before submitting? Seems not difficult.

**Review by antonkast-google — APPROVED** [review]
Approved contingent on addressing concerns in email.

File: webui/src/app/quick-phrases/quick-phrases.component.ts (hunk 2 above)
Comment (caisq) [reply webui/src/app/quick-phrases/quick-phrases.component.ts:201]: Ye. There is some divergence in the code between my branch and the main branch. So it makes sense to add the logging call later when the divergence is removed by another PR.

File: webui/src/app/event-logger/event-logger-impl.ts (hunk 1 above)
Comment (caisq) [reply webui/src/app/event-logger/event-logger-impl.ts:501]: Replied in email.

**Pull request merged**